lib/seek/merging/person_merge.rb

```diff
@@ -0,0 +1,113 @@
+module Seek
+  module Merging
+    module PersonMerge
+      def merge(other_person)
+        Person.transaction do
+          merge_simple_attributes(other_person)
+          merge_annotations(other_person)
+          # Roles have to be merged before group_memberships
+          merge_associations(other_person, 'roles', %i[scope_type scope_id role_type_id], { person_id: id })
+          # Merging group_memberships deals with work_groups, programmes, institutions and projects
+          merge_associations(other_person, 'group_memberships', [:work_group_id], { person_id: id })
+          merge_associations(other_person, 'subscriptions', [:subscribable_id], { person_id: id })
+          merge_associations(other_person, 'dependent_permissions', [:policy_id], { contributor_id: id })
+          merge_resources(other_person)
+          merge_user(other_person)
+
+          save!
+          other_person.reload # To prevent destruction of unlinked roles
+          other_person.destroy
+          ActivityLog.create!(action: 'MERGE-person',
+                              data: "Person with id #{other_person.id} was merged into person with id #{id}.")
+        end
+      end
+
+      private
+
+      # This attributes are directly copied from other_person if they are empty in the person to which its merging.
+      # first_letter is also updated
+      def simple_attributes
+        %i[
+          first_name
+          last_name
+          email
+          phone
+          skype_name
+          web_page
+          description
+          avatar_id
+          orcid
+        ]
+      end
+
+      def merge_simple_attributes(other_person)
+        simple_attributes.each do |attribute|
+          send("#{attribute}=", other_person.send(attribute)) if send(attribute).nil?
+        end
+        update_first_letter
+      end
+
+      def annotation_types
+        %w[
+          expertise
+          tools
+        ]
+      end
+
+      def merge_annotations(other_person)
```

Comment:
as mentioned in the call, also need to update the things annotated by the person.

you can get these with `other_person.user.annotations_by` ( a little different because they are linked to User instead of Person), and the update the source_ids to the new person.user.id checking for duplicates

Reply:
Resolved in [35536fa](https://github.com/seek4science/seek/pull/1683/commits/35536fa5900e158e29735c19d158090849b9d7a0).
The requirement might make sense... what it does at the moment is to ignore the other_person.user info if one or the other user are non existent...